packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol

```diff
@@ -0,0 +1,211 @@
+  // input & output currencies are the addresses of the ERC20 contracts OR the sha3("currency code")
+  mapping(address => mapping(address => address)) public allAggregators;
+
+  // rate must have been updated before the last... 10min
+  uint256 public maxTimestampDeltaAcceptable = 600;
+
+  // declare a new aggregator
+  event UpdateAggregator(address _input, address _output, address _aggregator);
+
+  /**
+    * @notice Update an aggregator
+    * @param _input address representing the input currency
+    * @param _output address representing the output currency
+    * @param _aggregator address of the aggregator contract
+  */
+  function updateAggregator(address _input, address _output, address _aggregator)
+    external
+    onlyWhitelistAdmin
+  {
+    allAggregators[_input][_output] = _aggregator;
+    emit UpdateAggregator(_input, _output, _aggregator);
+  }
+
+  /**
+    * @notice Update a list of aggregators
+    * @param _inputs list of addresses representing the input currencies
+    * @param _outputs list of addresses representing the output currencies
+    * @param _aggregators list of addresses of the aggregator contracts
+  */
+  function updateListAggregators(address[] calldata _inputs, address[] calldata _outputs, address[] calldata _aggregators)
+    external
+    onlyWhitelistAdmin
+  {
+    require(_inputs.length == _outputs.length, "arrays must have the same length");
+    require(_inputs.length == _aggregators.length, "arrays must have the same length");
+
+    // For every conversions of the path
+    for (uint i; i < _inputs.length; i++) {
+      allAggregators[_inputs[i]][_outputs[i]] = _aggregators[i];
+      emit UpdateAggregator(_inputs[i], _outputs[i], _aggregators[i]);
+    }
+  }
+
+  /**
+  * @notice Computes the conversion from an amount through a list of conversion
+  * @param _amountIn Amount to convert
+  * @param _path List of addresses representing the currencies for the conversions
+  * @return result the result after all the conversions
+  */
+  function getConversion(
+    uint256 _amountIn,
+    address[] calldata _path
+  )
+    external
+    view
+    returns (uint256 result, uint256 oldestTimestampRate)
+  {
+    (uint256 rate, uint256 oldestTimestampRateTemp, uint256 decimals) = getRate(_path);
+
+    // initialize the result
+    result = _amountIn.mul(rate).div(decimals);
+
+    oldestTimestampRate = oldestTimestampRateTemp;
+  }
+
+  /**
+  * @notice Computes the rate from a list of conversion
+  * @param _path List of addresses representing the currencies for the conversions
+  * @return result the result after all the conversions, the oldest rate of the path and the decimals (always 1e18)
+  */
+  function getRate(
+    address[] memory _path
+  )
+    public
+    view
+    returns (uint256 result, uint256 oldestTimestampRate, uint256 decimals)
+  {
+    // initialize the result with 1e18 decimals (for more precision)
+    result = DECIMALS;
+    decimals = DECIMALS;
```

Comment:
if I get this right, you are converting currencies that have less than 18 decimals (USDT for instance) later in the code. What's the point of returning this, if it's always 18?

Reply:
I feel the idea was to update this later, with `result` (so we know the amount of decimals we should use), but it has been forgotten.

Reply:
We can open a discussion about it. 
I wanted to add it in case later the decimals change regarding the currencies but we want to keep the same interface.